From bb486a1f679a1324ff18a59bceceacf37bbd05e6 Mon Sep 17 00:00:00 2001
From: Avinash M N <avimn@amd.com>
Date: Wed, 19 Apr 2017 08:57:54 +0530
Subject: [PATCH 095/104] drm/amdgpu: Revert page table updating using CPU

ROCm1.5 does not incorporate the code to update page tables using CPU.
This patch also cleans up some irrelevant code related to page table update

Signed-off-by: Avinash M N <avimn@amd.com>
---
 drivers/gpu/drm/amd/amdgpu/amdgpu.h              |   3 -
 drivers/gpu/drm/amd/amdgpu/amdgpu_amdkfd_gpuvm.c |   3 -
 drivers/gpu/drm/amd/amdgpu/amdgpu_drv.c          |   8 -
 drivers/gpu/drm/amd/amdgpu/amdgpu_vm.c           | 314 +----------------------
 drivers/gpu/drm/amd/amdgpu/amdgpu_vm.h           |  15 --
 5 files changed, 1 insertion(+), 342 deletions(-)

diff --git a/drivers/gpu/drm/amd/amdgpu/amdgpu.h b/drivers/gpu/drm/amd/amdgpu/amdgpu.h
index 0575524..0649771 100644
--- a/drivers/gpu/drm/amd/amdgpu/amdgpu.h
+++ b/drivers/gpu/drm/amd/amdgpu/amdgpu.h
@@ -90,9 +90,6 @@ extern int amdgpu_vm_size;
 extern int amdgpu_vm_block_size;
 extern int amdgpu_vm_fault_stop;
 extern int amdgpu_vm_debug;
-#if defined(CONFIG_HSA_AMD) || defined(CONFIG_HSA_AMD_MODULE)
-extern int amdgpu_vm_update_context;
-#endif
 extern int amdgpu_dal;
 extern int amdgpu_sched_jobs;
 extern int amdgpu_sched_hw_submission;
diff --git a/drivers/gpu/drm/amd/amdgpu/amdgpu_amdkfd_gpuvm.c b/drivers/gpu/drm/amd/amdgpu/amdgpu_amdkfd_gpuvm.c
index 0309e75..a3ad234 100644
--- a/drivers/gpu/drm/amd/amdgpu/amdgpu_amdkfd_gpuvm.c
+++ b/drivers/gpu/drm/amd/amdgpu/amdgpu_amdkfd_gpuvm.c
@@ -1380,7 +1380,6 @@ int amdgpu_amdkfd_gpuvm_create_process_vm(struct kgd_dev *kgd, void **vm,
 	int ret;
 	struct amdkfd_vm *new_vm;
 	struct amdkfd_process_info *info;
-	bool vm_update_using_cpu;
 	struct amdgpu_device *adev = get_amdgpu_device(kgd);
 
 	BUG_ON(kgd == NULL);
@@ -1391,8 +1390,6 @@ int amdgpu_amdkfd_gpuvm_create_process_vm(struct kgd_dev *kgd, void **vm,
 		return -ENOMEM;
 
 	/* Initialize the VM context, allocate the page directory and zero it */
-	vm_update_using_cpu = !!(amdgpu_vm_update_context &
-				AMDGPU_VM_USE_CPU_UPDATE_FOR_COMPUTE_MASK);
 	ret = amdgpu_vm_init(adev, &new_vm->base, true);
 	if (ret != 0) {
 		pr_err("Failed init vm ret %d\n", ret);
diff --git a/drivers/gpu/drm/amd/amdgpu/amdgpu_drv.c b/drivers/gpu/drm/amd/amdgpu/amdgpu_drv.c
index d9bcb30..0d26dc3 100644
--- a/drivers/gpu/drm/amd/amdgpu/amdgpu_drv.c
+++ b/drivers/gpu/drm/amd/amdgpu/amdgpu_drv.c
@@ -90,9 +90,6 @@ int amdgpu_vm_size = 64;
 int amdgpu_vm_block_size = -1;
 int amdgpu_vm_fault_stop = 0;
 int amdgpu_vm_debug = 0;
-#if defined(CONFIG_HSA_AMD) || defined(CONFIG_HSA_AMD_MODULE)
-int amdgpu_vm_update_context = 2;
-#endif
 int amdgpu_vram_page_split = 1024;
 int amdgpu_exp_hw_support = 0;
 int amdgpu_dal = -1;
@@ -177,11 +174,6 @@ module_param_named(vm_fault_stop, amdgpu_vm_fault_stop, int, 0444);
 MODULE_PARM_DESC(vm_debug, "Debug VM handling (0 = disabled (default), 1 = enabled)");
 module_param_named(vm_debug, amdgpu_vm_debug, int, 0644);
 
-#if defined(CONFIG_HSA_AMD) || defined(CONFIG_HSA_AMD_MODULE)
-MODULE_PARM_DESC(vm_update_context, "VM update using CPU on large BAR. (0 = never, 1 = Graphics only, 2 = Compute only (default), 3 = Both");
-module_param_named(vm_update_context, amdgpu_vm_update_context, int, 0444);
-#endif
-
 MODULE_PARM_DESC(vram_page_split, "Number of pages after we split VRAM allocations (default 1024, -1 = disable)");
 module_param_named(vram_page_split, amdgpu_vram_page_split, int, 0444);
 
diff --git a/drivers/gpu/drm/amd/amdgpu/amdgpu_vm.c b/drivers/gpu/drm/amd/amdgpu/amdgpu_vm.c
index 0683276..c3be10b 100644
--- a/drivers/gpu/drm/amd/amdgpu/amdgpu_vm.c
+++ b/drivers/gpu/drm/amd/amdgpu/amdgpu_vm.c
@@ -59,10 +59,6 @@ struct amdgpu_pte_update_params {
 	struct amdgpu_device *adev;
 	/* address where to copy page table entries from */
 	uint64_t src;
-#if defined(CONFIG_HSA_AMD) || defined(CONFIG_HSA_AMD_MODULE)
-	/* DMA addresses to use for mapping */
-	dma_addr_t *pages_addr;
-#endif
 	/* indirect buffer to fill with commands */
 	struct amdgpu_ib *ib;
 	/* Function which actually does the update */
@@ -71,48 +67,8 @@ struct amdgpu_pte_update_params {
 		     uint32_t flags);
 	/* indicate update pt or its shadow */
 	bool shadow;
-#if defined(CONFIG_HSA_AMD) || defined(CONFIG_HSA_AMD_MODULE)	
-	/* Kernel pointer of PD/PT BO that needs to be updated */
-	void *kptr;
-#endif	
 };
 
-#if defined(CONFIG_HSA_AMD) || defined(CONFIG_HSA_AMD_MODULE)
-/**
- * write_pte_using_cpu - update the page tables using CPU
- *
- * @adev: amdgpu_device pointer
- * @pte_update_params: see structure definition
- * @pe_start: offset from start of the page table
- * @addr: dst addr to write into pe
- * @count: number of page entries to update
- * @incr: increase next addr by incr bytes
- * @flags: hw access flags
- *
- */
-static void write_pte_using_cpu(struct amdgpu_device *adev,
-				struct amdgpu_pte_update_params *params,
-				uint64_t pe_start, uint64_t addr,
-				unsigned count, uint32_t incr,
-				uint32_t flags)
-{
-	unsigned int i;
-	uint64_t page_idx, value;
-
-	page_idx = pe_start / 8;
-
-	for (i = 0; i < count; i++) {
-		value = amdgpu_vm_map_gart(params->pages_addr, addr);
-		amdgpu_gart_set_pte_pde(adev, params->kptr,
-			page_idx, value, flags);
-		addr += incr;
-		page_idx++;
-	}
-	mb();
-	amdgpu_gart_flush_gpu_tlb(adev, 0);
-}
-#endif
-
 /**
  * amdgpu_vm_num_pde - return the number of page directory entries
  *
@@ -550,14 +506,7 @@ static void amdgpu_vm_do_set_ptes(struct amdgpu_pte_update_params *params,
 {
 	trace_amdgpu_vm_set_ptes(pe, addr, count, incr, flags);
 
-#if !defined(CONFIG_HSA_AMD) && !defined(CONFIG_HSA_AMD_MODULE)
 	if (count < 3) {
-#else
-	if (params->kptr) {
-		write_pte_using_cpu(params->adev, params, pe, addr,
-			count, incr, flags);
-	} else if (count < 3) {
-#endif
 		amdgpu_vm_write_pte(params->adev, params->ib, pe,
 				    addr | flags, count, incr);
 
@@ -601,33 +550,15 @@ static void amdgpu_vm_do_copy_ptes(struct amdgpu_pte_update_params *params,
  * Look up the physical address of the page that the pte resolves
  * to and return the pointer for the page table entry.
  */
-#if defined(CONFIG_HSA_AMD) || defined(CONFIG_HSA_AMD_MODULE)
-uint64_t amdgpu_vm_map_gart(const dma_addr_t *pages_addr, uint64_t addr)
-#else
 static uint64_t amdgpu_vm_map_gart(const dma_addr_t *pages_addr, uint64_t addr)
-#endif
 {
 	uint64_t result;
 
-#if !defined(CONFIG_HSA_AMD) && !defined(CONFIG_HSA_AMD_MODULE)
 	/* page table offset */
 	result = pages_addr[addr >> PAGE_SHIFT];
 
 	/* in case cpu page size != gpu page size*/
 	result |= addr & (~PAGE_MASK);
-#else
-	if (pages_addr) {
-		/* page table offset */
-		result = pages_addr[addr >> PAGE_SHIFT];
-
-		/* in case cpu page size != gpu page size*/
-		result |= addr & (~PAGE_MASK);
-
-	} else {
-		/* No mapping required */
-		result = addr;
-	}
-#endif
 
 	result &= 0xFFFFFFFFFFFFF000ULL;
 
@@ -674,7 +605,6 @@ int amdgpu_vm_update_page_directory(struct amdgpu_device *adev,
 	memset(&params, 0, sizeof(params));
 	params.adev = adev;
 
-#if !defined(CONFIG_HSA_AMD) && !defined(CONFIG_HSA_AMD_MODULE)
 	pd_addr = amdgpu_bo_gpu_offset(vm->root.bo);
 	if (shadow) {
 		r = amdgpu_ttm_bind(&shadow->tbo, &shadow->tbo.mem);
@@ -690,45 +620,7 @@ int amdgpu_vm_update_page_directory(struct amdgpu_device *adev,
 	if (r)
 		return r;
 	params.ib = &job->ibs[0];
-#else
-	/* If is_vm_update_mode_cpu flag is set, then try to update PD table
-	 * using CPU. If failed, fallback to use SDMA
-	 */
-	if (vm->is_vm_update_mode_cpu) {
-		struct amdgpu_sync sync;
-		struct amdgpu_bo *pd = vm->root.bo;
 
-		r = amdgpu_bo_kmap(vm->root.bo,
-			&params.kptr);
-		if (r)
-			dev_warn(adev->dev, "PD table update using CPU failed. Fallback to SDMA\n");
-		else {
-			amdgpu_sync_create(&sync);
-			amdgpu_sync_resv(adev, &sync, pd->tbo.resv,
-				AMDGPU_FENCE_OWNER_VM);
-			amdgpu_sync_wait(&sync);
-			amdgpu_sync_free(&sync);
-		}
-	}
-	if (r || !vm->is_vm_update_mode_cpu) {
-		pd_addr = amdgpu_bo_gpu_offset(vm->root.bo);
-		if (shadow) {
-			r = amdgpu_ttm_bind(&shadow->tbo, &shadow->tbo.mem);
-			if (r)
-				return r;
-			shadow_addr = amdgpu_bo_gpu_offset(shadow);
-			ndw *= 2;
-		} else {
-			shadow_addr = 0;
-		}
-
-		r = amdgpu_job_alloc_with_ib(adev, ndw * 4, &job);
-		if (r)
-			return r;
-
-		params.ib = &job->ibs[0];
-	}
-#endif
 	/* walk over the address space and update the page directory */
 	for (pt_idx = 0; pt_idx <= vm->root.last_entry_used; ++pt_idx) {
 		struct amdgpu_bo *bo = vm->root.entries[pt_idx].bo;
@@ -793,7 +685,6 @@ int amdgpu_vm_update_page_directory(struct amdgpu_device *adev,
 				      count, incr, AMDGPU_PTE_VALID);
 	}
 
-	/* Submit IB if updating using SDMA */
 	if (params.ib) {
 		if (params.ib->length_dw == 0) {
 			amdgpu_job_free(job);
@@ -816,13 +707,7 @@ int amdgpu_vm_update_page_directory(struct amdgpu_device *adev,
 		fence_put(vm->last_dir_update);
 		vm->last_dir_update = fence_get(fence);
 		fence_put(fence);
-#if !defined(CONFIG_HSA_AMD) && !defined(CONFIG_HSA_AMD_MODULE)
-	}
-#else
-	} else if (params.kptr) {
-		amdgpu_bo_kunmap(vm->root.bo);
 	}
-#endif
 
 	return 0;
 
@@ -832,7 +717,7 @@ error_free:
 }
 
 /**
- * amdgpu_vm_update_ptes_sdma - make sure that page tables are valid
+ * amdgpu_vm_update_ptes - make sure that page tables are valid
  *
  * @params: see amdgpu_pte_update_params definition
  * @vm: requested vm
@@ -843,17 +728,10 @@ error_free:
  *
  * Update the page tables in the range @start - @end using SDMA
  */
-#if defined(CONFIG_HSA_AMD) || defined(CONFIG_HSA_AMD_MODULE)
-static void amdgpu_vm_update_ptes_sdma(struct amdgpu_pte_update_params *params,
-				  struct amdgpu_vm *vm,
-				  uint64_t start, uint64_t end,
-				  uint64_t dst, uint32_t flags)
-#else
 static void amdgpu_vm_update_ptes(struct amdgpu_pte_update_params *params,
 				  struct amdgpu_vm *vm,
 				  uint64_t start, uint64_t end,
 				  uint64_t dst, uint32_t flags)
-#endif
 {
 	const uint64_t mask = AMDGPU_VM_PTE_COUNT - 1;
 
@@ -930,96 +808,6 @@ static void amdgpu_vm_update_ptes(struct amdgpu_pte_update_params *params,
 		     AMDGPU_GPU_PAGE_SIZE, flags);
 }
 
-#if defined(CONFIG_HSA_AMD) || defined(CONFIG_HSA_AMD_MODULE)
-/*
- * amdgpu_vm_frag_ptes - add fragment information to PTEs
- *
- * @params: see amdgpu_pte_update_params definition
- * @vm: requested vm
- * @start: first PTE to handle
- * @end: last PTE to handle
- * @dst: addr those PTEs should point to
- * @flags: hw mapping flags
- */
-static void amdgpu_vm_frag_ptes(struct amdgpu_pte_update_params	*params,
-				struct amdgpu_vm *vm,
-				uint64_t start, uint64_t end,
-				uint64_t dst, uint32_t flags)
-{
-	/**
-	 * The MC L1 TLB supports variable sized pages, based on a fragment
-	 * field in the PTE. When this field is set to a non-zero value, page
-	 * granularity is increased from 4KB to (1 << (12 + frag)). The PTE
-	 * flags are considered valid for all PTEs within the fragment range
-	 * and corresponding mappings are assumed to be physically contiguous.
-	 *
-	 * The L1 TLB can store a single PTE for the whole fragment,
-	 * significantly increasing the space available for translation
-	 * caching. This leads to large improvements in throughput when the
-	 * TLB is under pressure.
-	 *
-	 * The L2 TLB distributes small and large fragments into two
-	 * asymmetric partitions. The large fragment cache is significantly
-	 * larger. Thus, we try to use large fragments wherever possible.
-	 * Userspace can support this by aligning virtual base address and
-	 * allocation size to the fragment size.
-	 */
-
-	/* SI and newer are optimized for 64KB */
-	uint64_t frag_flags = AMDGPU_PTE_FRAG(AMDGPU_LOG2_PAGES_PER_FRAG);
-	uint64_t frag_align = 1 << AMDGPU_LOG2_PAGES_PER_FRAG;
-
-	uint64_t frag_start = ALIGN(start, frag_align);
-	uint64_t frag_end = end & ~(frag_align - 1);
-
-	unsigned count;
-	uint32_t incr = AMDGPU_VM_PTE_COUNT * 8;
-	
-	/* system pages are non continuously */
-	if (params->src || params->pages_addr || !(flags & AMDGPU_PTE_VALID) ||
-	    (frag_start >= frag_end)) {
-
-		count = (end - start)/8;
-		if (params->kptr)
-			write_pte_using_cpu(params->adev, params, start, dst, count, incr, flags);
-		else
-			amdgpu_vm_update_ptes_sdma(params, vm, start, end, dst, flags);
-		return;
-	}
-
-	/* handle the 4K area at the beginning */
-	if (start != frag_start) {
-	
-		count = (frag_start - start)/8;
-		if (params->kptr)
-			write_pte_using_cpu(params->adev, params, start, dst, count, incr, flags);
-		else
-			amdgpu_vm_update_ptes_sdma(params, vm, start, frag_start,
-				      	dst, flags);
-		dst += (frag_start - start) * AMDGPU_GPU_PAGE_SIZE;
-	}
-
-	/* handle the area in the middle */
-	count = (frag_end - frag_start)/8;
-	if (params->kptr)
-		write_pte_using_cpu(params->adev, params, frag_start, dst, count, incr, flags);
-	else	
-		amdgpu_vm_update_ptes_sdma(params, vm, frag_start, frag_end, dst,
-		      flags | frag_flags);
-
-	/* handle the 4K area at the end */
-	if (frag_end != end) {
-		dst += (frag_end - frag_start) * AMDGPU_GPU_PAGE_SIZE;
-		count = (end - frag_end)/8;
-		if (params->kptr)
-			write_pte_using_cpu(params->adev, params, frag_end, dst, count, incr, flags);
-		else		
-			amdgpu_vm_update_ptes_sdma(params, vm, frag_end, end, dst, flags);
-	}
-}
-
-#else
-
 /*
  * amdgpu_vm_frag_ptes - add fragment information to PTEs
  *
@@ -1087,64 +875,6 @@ static void amdgpu_vm_frag_ptes(struct amdgpu_pte_update_params	*params,
 	}
 }
 
-#endif
-
-#if defined(CONFIG_HSA_AMD) || defined(CONFIG_HSA_AMD_MODULE)
-/**
- * amdgpu_vm_update_ptes_cpu - make sure that page tables are valid
- *
- * @adev: amdgpu_device pointer
- * @pte_update_params: see amdgpu_vm_update_params definition
- * @vm: requested vm
- * @start: start of GPU address range
- * @end: end of GPU address range
- * @dst: destination address to map to
- * @flags: mapping flags
- *
- * Update the page tables in the range @start - @end using CPU
- */
-static int amdgpu_vm_update_ptes_cpu(struct amdgpu_pte_update_params
-					*vm_update_params,
-				     struct amdgpu_vm *vm,
-				     uint64_t start, uint64_t end,
-				     uint64_t dst, uint32_t flags)
-{
-	const uint64_t mask = AMDGPU_VM_PTE_COUNT - 1;
-	int r;
-	uint64_t addr;
-	struct amdgpu_bo *pt = NULL;
-
-	/* walk over the address space and update the page tables */
-	for (addr = start; addr < end; ) {
-		uint64_t pt_idx = addr >> amdgpu_vm_block_size;
-		unsigned nptes;
-		uint64_t pe_start;
-
-		pt = vm->root.entries[pt_idx].bo;
-		r = amdgpu_bo_kmap(pt, &vm_update_params->kptr);
-		if (r)
-			return r;
-
-		if ((addr & ~mask) == (end & ~mask))
-			nptes = end - addr;
-		else
-			nptes = AMDGPU_VM_PTE_COUNT - (addr & mask);
-
-		pe_start = (addr & mask) * 8;
-
-		amdgpu_vm_frag_ptes(vm_update_params, vm,
-			pe_start, pe_start + 8 * nptes,
-			dst, flags);
-
-		addr += nptes;
-		dst += nptes * AMDGPU_GPU_PAGE_SIZE;
-
-		amdgpu_bo_kunmap(pt);
-	}
-	return 0;
-}
-#endif
-
 /**
  * amdgpu_vm_bo_update_mapping - update a mapping in the vm page table
  *
@@ -1183,38 +913,6 @@ static int amdgpu_vm_bo_update_mapping(struct amdgpu_device *adev,
 	params.adev = adev;
 	params.src = src;
 	ring = container_of(vm->entity.sched, struct amdgpu_ring, sched);
-#if defined(CONFIG_HSA_AMD) || defined(CONFIG_HSA_AMD_MODULE)
-
-
-	params.pages_addr = pages_addr;
-
-	/* If update flag is set to CPU, then try to update PT entries directly
-	 * by CPU. If failed, fallback to SDMA update.
-	 */
-	if (vm->is_vm_update_mode_cpu) {
-		struct amdgpu_sync sync;
-
-		amdgpu_sync_create(&sync);
-		r = amdgpu_sync_resv(adev, &sync, vm->root.bo->tbo.resv,
-			owner);
-		if (r) {
-			dev_warn(adev->dev, "CPU pte update failed. Fallback to SDMA\n");
-			goto fallback_sdma_update;
-		}
-		amdgpu_sync_wait(&sync);
-		amdgpu_sync_free(&sync);
-
-		r = amdgpu_vm_update_ptes_cpu(&params,
-			vm, start, last + 1, addr, flags);
-		if (r) {
-			dev_info(adev->dev, "CPU pte update failed. Fallback to sdma\n");
-			goto fallback_sdma_update;
-		}
-		return r;
-	}
-
-fallback_sdma_update:
-#endif
 
 	/* sync to everything on unmapping */
 	if (!(flags & AMDGPU_PTE_VALID))
@@ -1550,11 +1248,7 @@ int amdgpu_vm_clear_invalids(struct amdgpu_device *adev,
 	}
 	spin_unlock(&vm->status_lock);
 
-#if defined(CONFIG_HSA_AMD) || defined(CONFIG_HSA_AMD_MODULE)
-	if (bo_va && sync)
-#else
 	if (bo_va)
-#endif
 		r = amdgpu_sync_fence(adev, sync, bo_va->last_pt_update);
 
 	return r;
@@ -1689,13 +1383,7 @@ int amdgpu_vm_bo_map(struct amdgpu_device *adev,
 		r = amdgpu_bo_create(adev, AMDGPU_VM_PTE_COUNT * 8,
 				     AMDGPU_GPU_PAGE_SIZE, true,
 				     AMDGPU_GEM_DOMAIN_VRAM,
-#if defined(CONFIG_HSA_AMD) || defined(CONFIG_HSA_AMD_MODULE)
-				     vm->is_vm_update_mode_cpu ?
-					AMDGPU_GEM_CREATE_CPU_ACCESS_REQUIRED :
-					AMDGPU_GEM_CREATE_NO_CPU_ACCESS |
-#else
 				     AMDGPU_GEM_CREATE_NO_CPU_ACCESS |
-#endif
 				     AMDGPU_GEM_CREATE_SHADOW |
 				     AMDGPU_GEM_CREATE_VRAM_CONTIGUOUS |
 				     AMDGPU_GEM_CREATE_VRAM_CLEARED,
diff --git a/drivers/gpu/drm/amd/amdgpu/amdgpu_vm.h b/drivers/gpu/drm/amd/amdgpu/amdgpu_vm.h
index 3fed8a60..c5b32bb 100644
--- a/drivers/gpu/drm/amd/amdgpu/amdgpu_vm.h
+++ b/drivers/gpu/drm/amd/amdgpu/amdgpu_vm.h
@@ -70,16 +70,6 @@ struct amdgpu_bo_list_entry;
 #define AMDGPU_VM_FAULT_STOP_FIRST	1
 #define AMDGPU_VM_FAULT_STOP_ALWAYS	2
 
-/* amdgpu_vm_update_context - User parameter that controls how VM pde/pte are
- *  updated for Graphics and Compute. BIT0 controls Graphics and BIT1 Compute.
- *  BIT0 [= 0] Graphics updated by SDMA [= 1] by CPU
- *  BIT1 [= 0] Compute updated by SDMA [= 1] by CPU
- *
- *  NOTE: CPU update is supported only for Large BAR systems
- */
-#define AMDGPU_VM_USE_CPU_UPDATE_FOR_GRAPHICS_MASK (1 << 0)
-#define AMDGPU_VM_USE_CPU_UPDATE_FOR_COMPUTE_MASK (1 << 1)
-
 struct amdgpu_vm_pt {
 	struct amdgpu_bo	*bo;
 	uint64_t		addr;
@@ -123,8 +113,6 @@ struct amdgpu_vm {
 	u64                     client_id;
 
 #if defined(CONFIG_HSA_AMD) || defined(CONFIG_HSA_AMD_MODULE)
-	/* Flag to indicate if VM tables are updated by CPU or GPU (SDMA) */
-	bool			is_vm_update_mode_cpu : 1;
 	/* Flag to indicate if this is a KFD VM */
 	bool			is_kfd_vm : 1;
 #endif
@@ -199,9 +187,6 @@ int amdgpu_vm_grab_id(struct amdgpu_vm *vm, struct amdgpu_ring *ring,
 		      struct amdgpu_job *job);
 int amdgpu_vm_flush(struct amdgpu_ring *ring, struct amdgpu_job *job);
 void amdgpu_vm_reset_id(struct amdgpu_device *adev, unsigned vm_id);
-#if defined(CONFIG_HSA_AMD) || defined(CONFIG_HSA_AMD_MODULE)
-uint64_t amdgpu_vm_map_gart(const dma_addr_t *pages_addr, uint64_t addr);
-#endif
 int amdgpu_vm_update_page_directory(struct amdgpu_device *adev,
 				    struct amdgpu_vm *vm);
 int amdgpu_vm_clear_freed(struct amdgpu_device *adev,
-- 
2.7.4

